airflow/jobs/scheduler_job_runner.py

```diff
@@ -1751,6 +1946,19 @@ def _schedule_dag_run(
                     "The DAG disappeared before verifying integrity: %s. Skipping.", dag_run.dag_id
                 )
                 return callback
+
+            if (
+                dag_run.scheduled_by_job_id is not None
+                and dag_run.set_scheduled_by_job_id != self.job.id
```

Comment:
```suggestion
                dag_run.set_scheduled_by_job_id != self.job.id
```

Don't think we need the `not None` check.

Reply:
Also: `set_scheduled_by_job_id` is a function, not a property. I guess that means this isn't covered by unit tests?

```suggestion
                dag_run.scheduled_by_job_id != self.job.id
```

Reply:
> Don't think we need the not None check.

I gave it a default value of `None` on the `DagRun` constructor.

Reply:
>  and dag_run.set_scheduled_by_job_id != self.job.id

I don't know how I missed that. There are no unit tests at the moment but there are integration tests and there have been some failures which I'm debugging. This must be one of the issues.

Another issue is that the tests stopped capturing the task output. I pushed a commit with a fix for that.

I'll add some unit tests as well for the scheduler changes.

Reply:
I've added unit tests for the scheduler changes. I should have done that in the first place.